### ISSUE TITLE
running webpacked js: Error: ENOENT: no such file or directory, scandir '/protocol'

### ISSUE THREAD

**Issue by hexray-newbee** [dresende/node-modbus-pdu#4]
windows 10/centos 7.4
node@^8
webpack@^3
modbus-stream@0.41.0

`var files = fs.readdirSync(path.resolve(__dirname, "protocol"));` at https://github.com/node-modbus/pdu/blob/master/lib/Modbus.js#L31

```
fs.js:904
  return binding.readdir(pathModule._makeLong(path), options.encoding);
                 ^

Error: ENOENT: no such file or directory, scandir '/protocol'
    at Object.fs.readdirSync (fs.js:904:18)
```

**Comment by dresende** [comment]
It just lists the `protocol` folder inside `lib` folder.

Does `path.resolve(__dirname, "protocol")` return you a wrong path? Does it need a trailing slash?

**Comment by hexray-newbee** [comment]
I have created a repo to reproduce: https://github.com/hxl-dy/node-modbus-pdu-4

**Comment by dresende** [comment]
```sh
~/Desktop/node-modbus-pdu-4 master › npm run webpack

> node-modbus-pdu-4@1.0.0 webpack /Users/dresende/Desktop/node-modbus-pdu-4
> webpack

Hash: f5bfe266a34a76c384c9
Version: webpack 3.12.0
Time: 419ms
   Asset    Size  Chunks             Chunk Names
index.js  196 kB       0  [emitted]  index
  [37] ./node_modules/bindings 160 bytes {0} [optional] [built]
  [44] ./index.js 467 bytes {0} [built]
  [48] ./node_modules/modbus-pdu/lib/protocol ^\.\/.*$ 1.12 kB {0} [built]
    + 80 hidden modules

WARNING in ./node_modules/bindings/bindings.js
81:22-40 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/bindings/bindings.js
 @ ./node_modules/@serialport/bindings/lib/win32.js
 @ ./node_modules/@serialport/bindings/lib/index.js
 @ ./node_modules/modbus-stream/node_modules/serialport/lib/index.js
 @ ./node_modules/modbus-stream/lib/driver/serial.js
 @ ./node_modules/modbus-stream/lib/modbus.js
 @ ./index.js

WARNING in ./node_modules/bindings/bindings.js
81:43-53 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/bindings/bindings.js
 @ ./node_modules/@serialport/bindings/lib/win32.js
 @ ./node_modules/@serialport/bindings/lib/index.js
 @ ./node_modules/modbus-stream/node_modules/serialport/lib/index.js
 @ ./node_modules/modbus-stream/lib/driver/serial.js
 @ ./node_modules/modbus-stream/lib/modbus.js
 @ ./index.js
```

**Comment by hexray-newbee** [comment]
please run `npm test`, and then this error occurs.
`npm test` is equal to `node build/index.js`

**Comment by dresende** [comment]
I think I now understand the problem. You're using webpack to use the library for example client side. I'm not sure if that's possible, will need to read more about webpack and see if I can hint it into compiling this correctly.

**Comment by dresende** [comment]
It seems there's a `__dirname` option under the `node` section in your config that can be tweak to try to workaround this. For now it's impossible to me to solve this.

**Comment by dresende** [comment]
This has been fixed in #7
